### ISSUE TITLE
Feature request; use the "window scale factor" engine for scroll zooming

### ISSUE THREAD

**Issue by galfisk** [OpenRCT2/OpenRCT2#14299]
When zooming out with the scroll wheel, OpenRCT2 uses the original RCT downscaler, which drops quite a bit of detail when zooming out. The openRCT2 scaler factor engine is much better, so it'd be nice if that one could be used for zooming in and out instead, especially on high-res monitors. Some may like the original aesthetic, so this could be made an optional choice.

**Comment by janisozaur** [comment]
If I understood your pitch correctly, you would want the vehicles, peeps, etc. to not disappear when zooming out?

I believe it would not look pretty, but from technical point of view removing of `G1_FLAG_NO_ZOOM_DRAW` flag should probably do the trick.

**Comment by galfisk** [comment]
No, what I'd like would be for the game (but not the UI of course) to change size as if I had changed the "window scale factor" setting. This is because changing the scale factor is a much prettier way of making things bigger and smaller.

I can make some illustrative screenshots, if that helps.

**Comment by janisozaur** [comment]
So… a duplicate of https://github.com/OpenRCT2/OpenRCT2/issues/775 then?

**Comment by galfisk** [comment]
No (that one is about zooming in), but reading the comments, I see that mine is a duplicate of what's proposed for Retina displays in https://github.com/OpenRCT2/OpenRCT2/issues/775#issuecomment-394026535

**Comment by janisozaur** [comment]
I can't say I fully understand. I'd like to take your offer about creating some screenshots, that might explain it better

**Comment by galfisk** [comment]
Sure thing.
On my monitor, I enjoy the game best with a scale factor of 1.75:
![Original zoom level](https://user-images.githubusercontent.com/31153158/111075036-4df24680-84e6-11eb-9d7b-28b4d19c44f6.png)

If I zoom out by scrolling, while retaining the window scale factor of 1.75, it looks like this:
![Scroll wheel zoom -1](https://user-images.githubusercontent.com/31153158/111075108-acb7c000-84e6-11eb-8016-852f1a747907.png)

But I can achieve almost the same change in scale by setting the Window scale factor to 1.00
![Scaling zoom -1](https://user-images.githubusercontent.com/31153158/111075127-bfca9000-84e6-11eb-859d-d955f4501761.png)

If you compare the two screenshots, you'll see that a lot more detail is present in the second one. This is why I propose that when possible, this method is used for scaling the image when zooming out, rather then the one presently used. Of course, UI elements still need to retain the desired scale factor.

Here are some more comparisons, showing that the scale factor always preserves more detail than the scroll zoom:
Scroll zoom -2 at Window scale factor 1.75:
![Scroll wheel zoom -2](https://user-images.githubusercontent.com/31153158/111075271-80507380-84e7-11eb-933b-51b7e7329d75.png)
Compared to Window scale factor of 0.5:
![Scaling zoom -2](https://user-images.githubusercontent.com/31153158/111075284-93634380-84e7-11eb-8956-0423aa1da443.png)

And scroll zoom -3 at Window scale factor 1.75 (smallest zoom):
![Scroll wheel zoom -3](https://user-images.githubusercontent.com/31153158/111075311-b8f04d00-84e7-11eb-9497-2d49b86dbd36.png)
Compared to Window scale factor of 0.5 combined with scroll zoom of -1:
![Scaling zoom -2 and scroll wheel zoom -1](https://user-images.githubusercontent.com/31153158/111075321-cc9bb380-84e7-11eb-9c62-29968acfcf1b.png)

The higher scale factor one uses, the more this will preserve detail when zooming out. Owners of 4K or 8K screens would be even more pleased than I.

**Comment by galfisk** [comment]
A further suggestion would be to add a hotkey of Ctrl+scroll to change the actual Window scale factor. Not sure how useful of an idea it is, but it'd make finding the scale factor you like much easier (especially since the button keeps jumping out form under the mouse pointer when you change it by clicking).

**Comment by janisozaur** [comment]
Assigning a shortcut to window scale should already be possible

**Comment by galfisk** [comment]
You're right. Nevermind on that particular idea then.

**Comment by galfisk** [comment]
I see now that the screenshots when viewed in full size don't give the correct picture, because the screenshot feature of OpenRCT2 undoes the window scale factor and just maks a higher res picture, but looking at them as they appear in the post gives the right idea.

**Comment by IntelOrca** [comment]
@galfisk are you suggesting that we render the game 1:1 scale of the whole map then downscale to the window size? That would be very expensive, impractical in fact on the performance. We currently also currently scale the viewports separately from the UI.

**Comment by galfisk** [comment]
No.
Let me take it from the top.

1. The suggestion pertains to those who run the game at a Window scale factor of more than 1, because they have a high res monitor. The game becomes tiny on such monitors when using a scale factor of 1.
2. Sometimes you _do_ want the game to be tiny, which is why you scroll to zoom out. When you do that, detail is discarded. When you do that while using a high scale factor, the resulting low res image is upscaled, which looks bad.
3. As mentioned, high res monitors make the game tiny st scale factor 1, so I suggest that when zooming out, the scale factor (of the viewport only, and not the UI) is decreased in steps until it hits 1. This lets players zoom out without losing detail. There is no downscaling involved, just smaller or no upscaling.

If further zoom-out steps are required beyond scale factor 1, the current method which discards detail shoud be applied while the scale factor remains 1. Zooming back in should first restore the detail and then increase the scale factor of the viewport. 
Using this method should be optional, and unavailable if the game is already set to a scale factor of 1 or below.

**Comment by duncanspumpkin** [comment]
What you described is not possible to achieve with the way the scale factor works

**Comment by IntelOrca** [comment]
Yes, we need to the window scale factor to only affect the UI, not the viewports. Currently we can not do this due to the architecture of the engine. We are not sure when it will be feasible. @ZehMatt might have more details.
